### ISSUE TITLE
onstatechange type value is a descriptive string in DEV but an integer in production

### ISSUE THREAD

**Issue by jasonkester** [downshift-js/downshift#913]
In the argument passed to the onStateChange handler, descriptive names such as "__autocomplete_keydown_arrow_down__" are used in the .type field.  But when the same app is build for production, the .type field will return an integer value, such as 4 (for that particular one).

This means that in order to detect the type of state change in a handler, one must test for *both* those possibilities.  That seems surprising and non-optimal.

It's also intentional.  Here is the PR that introduces this bug:

https://github.com/downshift-js/downshift/pull/548/files/328c9ce2cfa0d6c3433ea2d3a56ce61eb494d3fc

I would suggest reverting that.

**Comment by silviuavram** [comment]
Hi! The whole point of that is to abstract the state change types. You should not care if it’s a string in development or a number in production or whatever we will change it to in the future. You just use stateChangeTypes.YourStateChange and that’s it.

**Comment by jasonkester** [comment]
Thanks for the quick response.  That'll certainly get me back up and running.

I'll reiterate a strong disagree on the practice of intentionally changing enum values between dev and prod, since it'll lead to a lot more of this same issue from people debugging into issues like this.  You could eliminate it entirely by simply using that "4" in dev as well if you're concerned about bundle size.

**Comment by lostpebble** [comment]
Oh my... I just ran into this issue.

Thank you @jasonkester for opening this and bringing this up or I would be scratching my head for HOURS I'm sure.

I also agree that this is not great practice. Sometimes people develop their apps by looking at responses inside certain functions etc., and from that we make deterministic choices. Well, they should be deterministic at least...

It was extremely obscure having my deployment suddenly stop working. I already spent a bunch of time meticulously making sure my service workers weren't inadvertently causing some bundle sync issues.

Yes, it is a case of not reading all the documentation from top to bottom. But I found the way its presented to be a little daunting, and when I'm hacking away, I just want to throw things in and be productive.
